backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
       });
     }
-    const statusCode = 200;
-    const shortened = Boolean(req.query.shortened);
-    return res.status(statusCode).json({
-      id: req.params.id,
-      status: statusCode,
-      shortened,
-    });
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Change the activity data
+      const statusCode = 200;
+      const shortened = Boolean(req.query.shortened);
+	  const currentDateTime = Date.now();
+	  getobj.title = req.body.title || getobj.title;
+	  getobj.description = req.body.description || getobj.description;
+	  getobj.startDateTime = req.body.startDateTime || getobj.startDateTime;
+	  getobj.endDateTime = req.body.endDateTime || getobj.endDateTime;
+	  getobj.lastEditDateTime = currentDateTime;
+	  
+	  //Ensure end time comes after start time
+	  if (getobj.endDateTime < getobj.startDateTime) {
+		return res.status(400).json({
+          error: 'endDateTime is less than startDateTime',
+        });
+	  }
+	  
+	  //Save it to the database
+	  await getobj.save();
+	  
+	  //Response
+	  res.status(statusCode).json({
+        activity: getobj,
+		shortened,
+      });
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
+  });
+  
+/**
+ * Deletes an activity by id
+ * @param {Hex} id - The id of the activity, must be only hexadecimal
+ */
+activity.route('/delete/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
+        error: 'Not authenticated',
+      });
+    }
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Delete it
+	  await Activity.deleteOne(getobj);
+	  
+	  //Response
+	  res.sendStatus(200);
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
   });
 
 /**
- * Searches posts by criteria specified in post data
- * TODO: Add search criteria
- * @body {Number} numPosts - The number of posts to return
+ * Searches activites by custom criteria
+ * @body {Integer} page - The page to send back (Items sent are based on page and pageSize)
+ * @body {Integer} pageSize - The number of activities to return
+ * @body {Integer} sort - The sorting method (see below)
+ * @body {Integer} reverse - Reverses the order of the sort
+ * @body {Integer} omitEnded - Omits activities that have already ended. Defaults to 0.
+ * @body {Integer} omitStarted - Omits activities that have already started. Defaults to 0.
```

Comment:
These parameters can be boolean values instead of integers since we are using JSON formatting for our requests